### PR TITLE
[CI] Enable builds using private repos

### PR DESCRIPTION
**Pull request by peterargue**
This PR adds support for building containers that references private repos using CI. This is most commonly used for security patches, but could also work for forks.

### CODE REVIEW

**Review by sjonpaulbrown — APPROVED** [review]

**Review by zhangchiqing — APPROVED** [review]

**Comment by peterargue** [comment]
@franc-crypt0-s3curity 
> maybe replace /root/ with ~ ?

I tried this originally, but it failed for some reason. I'm not totally sure why. Ultimately, we need to move towards using dedicated users in the dockerfile and can refactor this at that point.

> the Makefile could be made more maitainable (but maybe less readable?) if some of the --secret id=git_creds,env=GITHUB_CREDS --build-arg GOPRIVATE=$(GOPRIVATE) \ were factored in one variable ?

This is a good idea. We really need to refactor our image building code since there is a ton of duplication. I'm going to open an issue to tackle that work.

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -105,12 +105,14 @@ jobs:
     - name: Build/Push ${{ matrix.role }} images
       env:
         IMAGE_TAG: ${{ inputs.docker_tag }}
+        GITHUB_CREDS: "machine github.com login ${{ secrets.REPO_SYNC_USER }} password ${{ secrets.REPO_SYNC }}"
```
[the remark this answers is not in the document]
Comment (peterargue) [reply .github/workflows/builds.yml:108]: This is primarily used in the public repo for regular patch builds. I'm hoping have a single workflow that works for both.

In the public repo, none of the repos match the `GOPRIVATE` list, so the github creds are ignored.

**Review by gomisha — APPROVED** [review]

**Comment by peterargue** [comment]
bors merge

**Comment by peterargue** [comment]
bors retry

**Pull request merged**